components/script/dom/location.rs

```diff
@@ -216,7 +222,10 @@ impl LocationMethods for Location {
     // https://html.spec.whatwg.org/multipage/#dom-location-pathname
     fn SetPathname(&self, value: USVString) -> ErrorResult {
         self.check_same_origin_domain()?;
-        self.set_url_component(value, UrlHelper::SetPathname);
+        // Step 3: If copyURL's cannot-be-a-base-URL flag is set, terminate these steps.
```

Comment:
Actually, can we put this comment, and all others, above the `Ok(())` branch, because the comments all refer to the steps take when the condition is not met.